File: src/main/scala/com/github/mauricio/postgresql/parsers/BackendKeyDataParser.scala
package com.github.mauricio.postgresql.parsers

import org.jboss.netty.buffer.ChannelBuffer
import com.github.mauricio.postgresql.messages.backend.{ProcessData, Message}

/**
 * User: Maurício Linhares
 * Date: 2/28/12
 * Time: 11:13 PM
 */

object BackendKeyDataParser extends Decoder {

  override def parseMessage(b: ChannelBuffer): Message = {
    new ProcessData( b.readInt(), b.readInt() )
  }

}

File: postgresql-async/src/main/java/com/github/jasync/sql/db/postgresql/parsers/BackendKeyDataParser.kt
package com.github.jasync.sql.db.postgresql.parsers

import com.github.jasync.sql.db.postgresql.messages.backend.ProcessData
import com.github.jasync.sql.db.postgresql.messages.backend.ServerMessage
import io.netty.buffer.ByteBuf

object BackendKeyDataParser : MessageParser {

    override fun parseMessage(buffer: ByteBuf): ServerMessage {
        return ProcessData(buffer.readInt(), buffer.readInt())
    }
}
